spec/lib/extensions/database_configuration_spec.rb

```diff
@@ -12,7 +12,7 @@
   context "ERB in the template" do
     before do
       database_config = StringIO.new "---\nvalue: <%= 1 %>\n"
-      allow(database_config).to receive(:exist?).and_return(true)
+      database_config.define_singleton_method(:exist?) { true }
```

Comment:
This seems like an abnormality stemming from the fact that we're returning a StringIO when the code is expecting a Pathname.  It might not be worth it but we could always write out an actual file to a Tempfile and return that instead of mocking methods on a StringIO

Reply:
> we could always write out an actual file to a Tempfile and return that instead of mocking methods on a StringIO

Yeah, I would much prefer this approach. Right now this spec is not really simulating the actual behavior very well. I would rather fix that problem than hack a fix into the already incorrect implementation.

Reply:
It doesn't seem like the manner in which the `exist?` check passes is important for this test, though. It looks like it just wants to make sure the erb doesn't get executed,  so creating a tempfile, etc, for one spec feels like overkill.

Reply:
I feel like given that this is a pretty low priority change which doesn't have any timeline for merge it makes more sense to take the time and effort to reduce technical debt.

Additionally, even though it doesn't matter to the exact problem you're trying to solve, making the spec reflect reality more closely is an objective improvement.